prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -188,6 +194,72 @@ public async Task<IActionResult> UpdateSiteCompleted(int siteId)
             return NoContent();
         }
 
+        // PUT: api/Sites/5/adjudicator
+        /// <summary>
+        /// Add a site's assigned adjudicator.
+        /// </summary>
+        /// <param name="siteId"></param>
+        /// <param name="adjudicatorId"></param>
+        [HttpPut("{siteId}/adjudicator", Name = nameof(SetSiteAdjudicator))]
+        [Authorize(Policy = AuthConstants.ADMIN_POLICY)]
+        [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<Site>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<Site>> SetSiteAdjudicator(int siteId, [FromQuery] int? adjudicatorId)
+        {
+            var site = await _siteService.GetSiteAsync(siteId);
+
+            if (site == null)
+            {
+                return NotFound(ApiResponse.Message($"Site not found with id {siteId}."));
+            }
+
+            if ((adjudicatorId.HasValue && !await _adminService.AdminExistsAsync(adjudicatorId.Value)))
+            {
+                return NotFound(ApiResponse.Message($"Admin not found with id {adjudicatorId.Value}."));
+            }
+
+            Admin admin = (adjudicatorId.HasValue)
+                ? await _adminService.GetAdminAsync(adjudicatorId.Value)
+                : await _adminService.GetAdminAsync(User.GetPrimeUserId());
+
```

Comment:
At this point, if no adjudicator ID was specified, and the current user does not have an Admin object, GetAdmin will return null. Is the intention to assign null to the site's Adjudicator in that case? If not, we can remove the check on line 219 and check if admin == null here.

Reply:
The assumption was that with the `ADMIN_POLICY` in place that they must have an `Admin` object.  I wouldn't have thought could make it this far without one.